src/main/scala/lc/server/Server.scala

```diff
@@ -24,14 +24,19 @@ class Server(port: Int) {
   }
 
   private val eqPattern = java.util.regex.Pattern.compile("=")
-  private def getPathParameter(ex: HttpExchange): String = {
+  private def getPathParameter(ex: HttpExchange): Either[String, Error] = {
     try {
       val query = ex.getRequestURI.getQuery
-      eqPattern.split(query)(1)
+      val param = eqPattern.split(query)
+      if(param(0) == "id"){
+        Left(param(1))
+      } else {
+        Right(Error(ErrorMessageEnum.INVALID_PARAM.toString + "=> id"))
```

Comment:
* By convention, Left should be failure, Right should be success.
* You needn't wrap error message into Error here. You can simply say Right(string) instead of Right(Error(string))